source/test-suite/Main.hs

```diff
@@ -143,13 +152,42 @@ spec = H.describe "Database.PostgreSQL.Simple.Interval" $ do
         let actual = Attoparsec.parseOnly I.parse input
         actual `H.shouldBe` Right interval
 
+  H.describe "integration" $ do
+    Monad.forM_ intervalStyles $ \(style, field) -> do
+      H.describe ("with style " <> show style) $ do
+        Monad.forM_ examples $ \example -> do
+          H.it ("round trips " <> show (field example)) $ do
+            Postgres.withConnect Postgres.defaultConnectInfo $ \connection -> do
+              let interval = exampleInterval example
+              result <- Exception.try . Postgres.withTransaction connection $ do
+                Monad.void $ Postgres.execute connection "set local intervalstyle = ?" [style]
+                Postgres.query connection "select ?" [interval]
+              case result of
+                Right actual -> actual `H.shouldBe` [Postgres.Only interval]
+                Left somePostgresqlException -> do
+                  rows <- Postgres.query_ connection "select version()"
```

Comment:
This should probably use [`serverVersion`](https://hackage.haskell.org/package/postgresql-libpq-0.10.2.0/docs/Database-PostgreSQL-LibPQ.html#v:serverVersion) instead.